flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -375,6 +375,12 @@ public void registerTaskManager(final TaskExecutorConnection taskExecutorConnect
 		if (taskManagerRegistrations.containsKey(taskExecutorConnection.getInstanceID())) {
 			reportSlotStatus(taskExecutorConnection.getInstanceID(), initialSlotReport);
 		} else {
+			if (getNumberRegisteredSlots() + Math.max(getNumberPendingTaskManagerSlots(), numSlotsPerWorker) > maxSlotNum) {
+				LOG.warn("The total number of slots exceeds the max limitation, release the excess resource.");
+				resourceActions.releaseResource(taskExecutorConnection.getInstanceID(), new FlinkException("The total number of slots exceeds the max limitation."));
+				return;
+			}
```

Comment:
Not sure about checking on task executors' registering.
I think the max slot limit should not take effect for standalone clusters, because the number of task executors launched is not controlled by RM. We should also mention that in the config option description.

Reply:
I think we could edit the contract of `ResourceActions#releaseResource`, let it return whether the release action succeeds. For `StandaloneResourceManager`, it would return false. In this case, it probably makes sense to continue to register the `reportedSlots`. So that, the slot manager need not be aware of whether it's running in a Standalone- or YarnResourceManager. WDYT?

Reply:
It seems to me that this approach implicitly assumes that `releaseResource` returns false means running on a standalone cluster, which is not always true. Without this assumption, continuing registering the slot when the release action failed seems quite against intuition.

Instead of let the slot manager behaves differently according to the release action result, I would suggest to simply pass in infinite large values for the max limits when creating the slot manager. In this way, slot manager should not need to behave differently on standalone and active resource managers.